pyV2DL3/vegas/EffectiveAreaFile.py

```diff
@@ -0,0 +1,140 @@
+        cut_searches = [
+            "ThetaSquareUpper",
+            "MeanScaledWidthLower", "MeanScaledWidthUpper",
+            "MaxHeightLower", "MaxHeightUpper",
+            "FoVCutUpper", "FoVCutLower",
+        ]
+
+        # Initialize corresponding class variables and default values
+        self.theta_square_upper = None
+        self.msw_lower = float('-inf')
+        self.msw_upper = float('inf')
+        self.max_height_lower = None
+        self.max_height_upper = None
+        self.fov_cut_lower = float('-inf')
+        self.fov_cut_upper = float('inf')
+
+        # Now load the cuts params
+        self.__load_cuts_info__(cut_searches)
+
+        # Build indexes for IRFs
+        self.axis_dict, self.index_dict = self.__build_index__()
+
+    """
+    Build az, zen, noise, and offset indexes for this file.
+    """
+    def __build_index__(self):
+        manager = self.manager
+        axis = ["Azimuth", "Zenith", "Noise"]
+        if len(manager.fEffectiveAreas) <= 0:
+            raise Exception("No effective areas! ")
+        index_check = manager.fEffectiveAreas.at(0).fDimensionNames
+        for k in axis:
+            if k not in index_check:
+                raise Exception("IRF missing axis: {}".format(k))
+        index_dict = {"Index": []}
+
+        for i, ea in enumerate(manager.fEffectiveAreas):
+            index_dict["Index"].append(i)
+            for name, val in zip(ea.fDimensionNames, ea.fDimensionValues):
+                if name not in index_dict:
+                    index_dict[name] = []
+                else:
+                    index_dict[name].append(val)
+
+        # Deal with AbsoluteOffset
+        if "AbsoluteOffset" not in index_check:
+            logger.info("No offset axis available from file. Use 0.5 deg as default.")
+            index_dict["AbsoluteOffset"] = []
+            for _ in range(len(index_dict["Index"])):
+                index_dict["AbsoluteOffset"].append(0.5)
+
+        # Validate Completeness
+        axis_dict = {}
+        check_num = 1
+
+        for k in axis + ["AbsoluteOffset"]:
+            check_num *= len(np.unique(index_dict[k]))
+            axis_dict[k] = np.sort(np.unique(index_dict[k]))
+            if len(axis_dict[k]) < 2 and k != "AbsoluteOffset":
+                raise Exception("{} Axis need to have more than two values".format(k))
+        return axis_dict, index_dict
+
+    def get_safe_energy(self, az, ze, noise):
+        manager = self.manager
+        effectiveAreaParameters = ROOT.VAEASimpleParameterData()
+        effectiveAreaParameters.fAzimuth = az
+        effectiveAreaParameters.fZenith = ze
+        effectiveAreaParameters.fNoise = noise
+
+        effectiveAreaParameters.fOffset = 0.5
+        effectiveAreaParameters = manager.getVectorParamsFromSimpleParameterData(
+            effectiveAreaParameters
+        )
+        minEnergy, maxEnergy = c_float(), c_float()
+        # Is it the right way ? what does the offset here provide ?
```

Comment:
Not sure if this questions requires an answer before merging.

Reply:
Interestingly, this is not my comment. This function and comment are from the IRFLoader class that I refactored into EffectiveAreaFile.py. This has been in the code for a few years I believe.

Reply:
I think you can ditch the comment as this looks like the correct method, at least the way it's done in VEGAS. I guess my only question would be, is it your intention to only support 0.5 degree wobble offset as is currently done here? As your offset value is hardcoded, you might want to explicitly comment on this somewhere.

Reply:
I will remove this comment and add the other, as well as open a github issue regarding offset support. Thanks!